src/Http/Controllers/CategoryController.php

```diff
@@ -10,161 +14,31 @@ public function show(int $id)
         foreach ($this->getCategories() as $category) {
             if ($category['id'] === $id) {
                 view("showCategory", [
-                    "products" => $category['Products'],
-                    "categoryName" => $category['CategoryName'],
+                    "products" => $this->getProducts($id),
+                    "category" => $this->getCategory($id),
                     "categories" => $this->getCategories()
                 ]);
```

Comment:
No need to get all categories and compare each category ID. This degrades the performance. Just get the category by ID with one query and another query to get its products.
```php
view("showCategory", [
  "products" => $this->getProducts($id),
  "category" => $this->getCategory($id),
  "categories" => $this->getCategories()
]);
```
But you need to handle whether the passed category ID is valid or not found (404)?